src/util/db/dbconnectionpool.h

```diff
@@ -0,0 +1,80 @@
+#ifndef MIXXX_DBCONNECTIONPOOL_H
+#define MIXXX_DBCONNECTIONPOOL_H
+
+
+#include <QAtomicInt>
+#include <QThreadStorage>
+
+#include "util/db/dbconnection.h"
+#include "util/memory.h"
+#include "util/assert.h"
+
+
+namespace mixxx {
+
+class DbConnectionPool;
+typedef std::shared_ptr<DbConnectionPool> DbConnectionPoolPtr;
+
+class DbConnectionPool final {
+  public:
+    static DbConnectionPoolPtr create(
+            const DbConnection::Params& params,
```

Comment:
I think this schould be a kind of singleton to avoid two classes using the static data.